packages/devtools_app/lib/src/screens/memory/panes/chart/chart_control_pane.dart

```diff
@@ -0,0 +1,227 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../../../../analytics/analytics.dart' as ga;
+import '../../../../analytics/constants.dart' as analytics_constants;
+import '../../../../primitives/auto_dispose_mixin.dart';
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/theme.dart';
+import '../../../../shared/utils.dart';
+import '../../memory_controller.dart';
+import '../../primitives/ui.dart';
+import '../../shared/constants.dart';
+import 'chart_pane_controller.dart';
+import 'interval_dropdown.dart';
+import 'legend.dart';
+
+class ChartControlPane extends StatefulWidget {
+  const ChartControlPane({Key? key, required this.chartController})
+      : super(key: key);
+  final MemoryChartPaneController chartController;
+
+  @override
+  State<ChartControlPane> createState() => _ChartControlPaneState();
+}
+
+class _ChartControlPaneState extends State<ChartControlPane>
+    with
+        ProvidedControllerMixin<MemoryController, ChartControlPane>,
+        AutoDisposeMixin {
+  OverlayEntry? _legendOverlayEntry;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) return;
+
+    addAutoDisposeListener(controller.legendVisibleNotifier, () {
+      setState(() {
+        if (controller.isLegendVisible) {
+          ga.select(
+            analytics_constants.memory,
+            analytics_constants.memoryLegend,
+          );
+
+          _showLegend(context);
+        } else {
+          _hideLegend();
+        }
+      });
+    });
+
+    // Refresh legend if android chary visibility changed.
+    addAutoDisposeListener(controller.isAndroidChartVisibleNotifier, () {
+      setState(() {
+        if (controller.isLegendVisible) {
+          // Recompute the legend with the new traces now visible.
+          _hideLegend();
+          _showLegend(context);
+        }
+      });
+    });
+  }
+
+  void _onPause() {
+    ga.select(analytics_constants.memory, analytics_constants.pause);
+    controller.pauseLiveFeed();
+  }
+
+  void _onResume() {
+    ga.select(analytics_constants.memory, analytics_constants.resume);
+    controller.resumeLiveFeed();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const buttonWidth = 110.0;
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        ValueListenableBuilder<bool>(
+          valueListenable: controller.paused,
+          builder: (context, paused, _) => Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              PauseButton(
+                minScreenWidthForTextBeforeScaling:
+                    minScreeWidthToKeepButtonsSmall,
```

Comment:
instead of using this minScreeWidthToKeepButtonsSmall property, just make an icon button. Feel free to create factory contructor PauseButton.iconOnly and ResumeButton.iconOnly.